src/pages/home/report/ReportActionCompose/SuggestionMention.tsx

```diff
@@ -56,6 +57,33 @@ type SuggestionPersonalDetailsList = Record<
     | null
 >;
 
+function getDisplayName(p: PersonalDetails) {
+    const displayNameFromAccountID = ReportUtils.getDisplayNameForParticipant(p.accountID);
+    if (!displayNameFromAccountID) {
+        return p.login?.length ? p.login : '';
+    }
+    return displayNameFromAccountID;
+}
+
+/**
+ * the comparison function used to determine which user will come first in the sorted list
+ * generally, smaller weight means will come first, and if the weight is the same, we'll sort based on displayName/login and accountID
+ */
+function compareUserInList(first: PersonalDetails & {weight: number}, second: PersonalDetails & {weight: number}) {
+    // first, we should sort by weight
+    if (first.weight !== second.weight) {
+        return first.weight - second.weight;
+    }
+
+    // next we sort by display name
+    const displayNameLoginOrder = localeCompare(getDisplayName(first), getDisplayName(second));
+    if (displayNameLoginOrder !== 0) {
+        return displayNameLoginOrder;
+    }
+    // then fallback on accountID as the final sorting criteria.
```

Comment:
The jsdoc already explains this, and this says what the code is doing, which is obvious from reading the code. Since these comments don't follow our style guide (should use capital letters on the first word), I suggest removing them
```suggestion
    if (first.weight !== second.weight) {
        return first.weight - second.weight;
    }
    const displayNameLoginOrder = localeCompare(getDisplayName(first), getDisplayName(second));
    if (displayNameLoginOrder !== 0) {
        return displayNameLoginOrder;
    }
```